src/hotspot/share/gc/g1/g1FullCollector.cpp

```diff
@@ -225,15 +228,20 @@ void G1FullCollector::complete_collection() {
   _heap->print_heap_after_full_collection(scope()->heap_transition());
 }
 
-void G1FullCollector::update_attribute_table(HeapRegion* hr) {
-  if (hr->is_free()) {
+void G1FullCollector::update_attribute_table(HeapRegion* hr, bool force_pinned) {
+  if (force_pinned) {
+    // Pin high live ratio region
+    _region_attr_table.set_pinned(hr->hrm_index());
     return;
   }
+
```

Comment:
Since the `if ... else if ...` cascade has already been started below, please continue with it.